gmp-task-utils-generated.h

```diff
@@ -0,0 +1,2416 @@
+  o_ (o), m_ (m), r_ (r) {
+  }
+  virtual bool returns_value () const
+  {
+    return true;
+  }
+
+  void Run ()
+  {
+    *r_ = ((*o_).*m_) ();
+  }
+
+private:
+  C o_;
+  M m_;
+  R *r_;
+};
+
+
+
+// 1 arguments --
+template < typename M, typename A0 > class gmp_task_args_nm_1:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_nm_1 (M m, A0 a0):
+  m_ (m), a0_ (a0) {
+  }
+
+  void Run ()
+  {
+    m_ (a0_);
+  }
+
+private:
+  M m_;
+  A0 a0_;
+};
+
+
+
+// 1 arguments --
+template < typename M, typename A0, typename R > class gmp_task_args_nm_1_ret:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_nm_1_ret (M m, A0 a0, R * r):
+  m_ (m), r_ (r), a0_ (a0) {
+  }
+  virtual bool returns_value () const
+  {
+    return true;
+  }
+
+  void Run ()
+  {
+    *r_ = m_ (a0_);
+  }
+
+private:
+  M m_;
+  R *r_;
+  A0 a0_;
+};
+
+
+
+// 1 arguments --
+template < typename C, typename M, typename A0 > class gmp_task_args_m_1:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_m_1 (C o, M m, A0 a0):
+  o_ (o), m_ (m), a0_ (a0) {
+  }
+
+  void Run ()
+  {
+    ((*o_).*m_) (a0_);
+  }
+
+private:
+  C o_;
+  M m_;
+  A0 a0_;
+};
+
+
+
+// 1 arguments --
+template < typename C, typename M, typename A0, typename R > class gmp_task_args_m_1_ret:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_m_1_ret (C o, M m, A0 a0, R * r):
+  o_ (o), m_ (m), r_ (r), a0_ (a0) {
+  }
+  virtual bool returns_value () const
+  {
+    return true;
+  }
+
+  void Run ()
+  {
+    *r_ = ((*o_).*m_) (a0_);
+  }
+
+private:
+  C o_;
+  M m_;
+  R *r_;
+  A0 a0_;
+};
+
+
+
+// 2 arguments --
+template < typename M, typename A0, typename A1 > class gmp_task_args_nm_2:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_nm_2 (M m, A0 a0, A1 a1):
+  m_ (m), a0_ (a0), a1_ (a1) {
+  }
+
+  void Run ()
+  {
+    m_ (a0_, a1_);
+  }
+
+private:
+  M m_;
+  A0 a0_;
+  A1 a1_;
+};
+
+
+
+// 2 arguments --
+template < typename M, typename A0, typename A1, typename R > class gmp_task_args_nm_2_ret:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_nm_2_ret (M m, A0 a0, A1 a1, R * r):
+  m_ (m), r_ (r), a0_ (a0), a1_ (a1) {
+  }
+  virtual bool returns_value () const
+  {
+    return true;
+  }
+
+  void Run ()
+  {
+    *r_ = m_ (a0_, a1_);
+  }
+
+private:
+  M m_;
+  R *r_;
+  A0 a0_;
+  A1 a1_;
+};
+
+
+
+// 2 arguments --
+template < typename C, typename M, typename A0, typename A1 > class gmp_task_args_m_2:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_m_2 (C o, M m, A0 a0, A1 a1):
+  o_ (o), m_ (m), a0_ (a0), a1_ (a1) {
+  }
+
+  void Run ()
+  {
+    ((*o_).*m_) (a0_, a1_);
+  }
+
+private:
+  C o_;
+  M m_;
+  A0 a0_;
+  A1 a1_;
+};
+
+
+
+// 2 arguments --
+template < typename C, typename M, typename A0, typename A1, typename R > class gmp_task_args_m_2_ret:public gmp_task_args_base
+{
+public:
+explicit gmp_task_args_m_2_ret (C o, M m, A0 a0, A1 a1, R * r):
```

Comment:
R *r

Reply:
This is generated code from Gecko that got formatted. I'll revert that, but it'll be nasty to review.